File: mypostbank-ui/src/app/services/models/repayment-detail-dto.ts
/* tslint:disable */
/* eslint-disable */
export interface RepaymentDetailDto {
  dueDate?: string;
  interestPayment?: number;
  month?: number;
  paymentAmount?: number;
  paymentStatus?: 'PAID' | 'NOT_PAID' | 'PARTIAL' | 'OVERDUE' | 'CANCELLED' | 'INSUFFICIENT_FUNDS';
  principalPayment?: number;
  remainingBalance?: number;
}

File: src/main/java/com/Spring/MyPostBank/DTOs/RepaymentDetailDTO.java
package com.Spring.MyPostBank.DTOs;

import com.Spring.MyPostBank.Enums.PaymentStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RepaymentDetailDTO {
    private Integer month;
    private LocalDate dueDate;
    private BigDecimal principalPayment;
    private BigDecimal interestPayment;
    private BigDecimal remainingBalance;
    private BigDecimal paymentAmount;
    private PaymentStatus paymentStatus;
}

